src/workload.rs

```diff
@@ -516,67 +456,15 @@ impl WorkloadManager {
     }
 }
 
-/// LocalClient serves as a local file reader alternative for XDS. This is intended for testing.
-struct LocalClient {
-    cfg: ConfigSource,
-    workloads: Arc<Mutex<WorkloadStore>>,
-}
-
-#[derive(Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct LocalWorkload {
-    #[serde(flatten)]
-    pub workload: Workload,
-    pub vips: HashMap<String, HashMap<u16, u16>>,
-}
-
-#[derive(Default, Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
-#[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct LocalConfig {
-    #[serde(default)]
-    pub workloads: Vec<LocalWorkload>,
-    #[serde(default)]
-    pub policies: Vec<Authorization>,
-    #[serde(default)]
-    pub services: Vec<Service>,
-}
-
-impl LocalClient {
-    #[instrument(skip_all, name = "local_client")]
-    async fn run(self) -> Result<(), anyhow::Error> {
-        // Currently, we just load the file once. In the future, we could dynamically reload.
-        let data = self.cfg.read_to_string().await?;
-        trace!("local config: {data}");
-        let r: LocalConfig = serde_yaml::from_str(&data)?;
-        let mut store = self.workloads.lock().unwrap();
-        let workloads = r.workloads.len();
-        let policies = r.policies.len();
-        for wl in r.workloads {
-            debug!("inserting local workload {}", &wl.workload.uid);
-            let vips = wl
-                .vips
-                .into_iter()
-                .map(|(k, v)| (k, PortList::from(v)))
-                .collect();
-            store.insert_workload(wl.workload, vips)?;
-        }
-        for rbac in r.policies {
-            store.insert_authorization(rbac);
-        }
-        for svc in r.services {
-            store.services.insert(svc);
-        }
-        info!(%workloads, %policies, "local config initialized");
-        Ok(())
-    }
-}
-
 /// WorkloadInformation wraps WorkloadStore, but is able to additionally request resources on-demand.
 /// It is designed to be cheap to clone.
 #[derive(serde::Serialize, Debug, Clone)]
 pub struct WorkloadInformation {
     #[serde(flatten)]
-    pub info: Arc<Mutex<WorkloadStore>>,
+    pub workloads: Arc<RwLock<WorkloadStore>>,
+
+    #[serde(flatten)]
+    pub services: Arc<RwLock<ServiceStore>>,
```

Comment:
FYI, I'm not loving this. Now there are 2 separate locks which need to be locked, which means if a call site needs both it has to lock in a consistent order. It would be better to have a single object that is lockable and contains all the various stores. I'm open to suggestions.